membership-attribute-service/app/services/AttributesMaker.scala

```diff
@@ -1,30 +1,41 @@
 package services
 
-import com.gu.memsub.Product
+import com.gu.memsub.{Benefit, Product}
 import com.gu.memsub.subsv2.SubscriptionPlan.AnyPlan
 import com.gu.memsub.subsv2.{GetCurrentPlans, Subscription}
 import com.typesafe.scalalogging.LazyLogging
 import models.Attributes
 import org.joda.time.LocalDate
+import com.github.nscala_time.time.OrderingImplicits.LocalDateOrdering
 
 class AttributesMaker extends LazyLogging {
 
   def attributes(identityId: String, subs: List[Subscription[AnyPlan]], forDate: LocalDate): Option[Attributes] = {
 
-    val groupedSubs: Map[Option[Product], List[Subscription[AnyPlan]]] = subs.groupBy(subscription => GetCurrentPlans(subscription, forDate).toOption.map(_.head.product))
-    val membershipSub = groupedSubs.getOrElse(Some(Product.Membership), Nil)
-    val contributionSub = groupedSubs.getOrElse(Some(Product.Contribution), Nil)
+    def getCurrentPlans(subscription: Subscription[AnyPlan]) = GetCurrentPlans(subscription, forDate).toOption
+    def getTopProduct(subscription: Subscription[AnyPlan]) = getCurrentPlans(subscription).map(_.head.product)
```

Comment:
not down to your change, but it's interesting that we're turning an error into an option (.toOption on the either) and an option into an exception (.head on a list).  Maybe subs with no plans are already filtered out, so .head won't fail.